zombies.js

```diff
@@ -79,6 +98,90 @@
  * @property {method} getMaxHealth         Returns private variable `maxHealth`.
  */
 
+function Player(name, health, strength, speed) {
+  this.name = name;
+  this.health = health;
+  this.strength = strength;
+  this.speed = speed;
+  this.isAlive = true;
+  this.equipped = false;
+  _pack = [];
+  _maxHealth = health
+}
+
+Player.prototype.getPack = function () {
+  return _pack
+}
+Player.prototype.getMaxHealth = function () {
+  return _maxHealth
+}
+Player.prototype.checkPack = function () {
+  console.log(this.getPack())
+}
+Player.prototype.takeItem = function (item) {
+  if (this.getPack().length < 3) {
+    this.getPack().push(item);
+    console.log(this.name + " picks up " + item.name)
+  } else {
+    console.log("Item cannot be stored. Pack is full");
+  }
+}
+Player.prototype.discardItem = function (item) {
+  let itemToBeRemoved = this.getPack().indexOf(item);
+  if (itemToBeRemoved === -1) {
+    console.log("Cannot be removed. Item not found");
+  } else {
+    this.getPack().splice(itemToBeRemoved, 1);
+    console.log(item.name + " was removed");
+    return true
+  }
+}
+Player.prototype.equip = function (itemToEquip) {
+  let itemInPack = this.getPack().indexOf(itemToEquip);
+  if (itemToEquip instanceof Weapon && itemInPack !== -1) {
+    if (this.equipped === false) {
+      this.equipped = itemToEquip;
+      this.discardItem(itemToEquip)
+    } else if (this.equipped !== false) {
+      this.getPack().splice(itemInPack, 1, this.equipped);
+      this.equipped = itemToEquip;
+      this.discardItem(itemToEquip)
+    } else {
+      console.log("Item not found / Item is not a weapon")
+    }
+  }
+}
+Player.prototype.eat = function (itemToEat) {
+  let foodInPack = this.getPack().indexOf(itemToEat);
+  if (itemToEat instanceof Food && foodInPack !== -1) {
+    if ((itemToEat.energy + this.health) > this.getMaxHealth()) {
+      this.health = this.getMaxHealth();
+      this.discardItem(itemToEat)
+    } else if ((itemToEat.energy + this.health) <= this.getMaxHealth()) {
+      this.health += itemToEat.energy;
+      this.discardItem(itemToEat)
```

Comment:
Can you think of a way to only call `this.discardItem(itemToEat)` once in this function.